src/UiPath.Workflow/Activities/ScriptingJitCompiler.cs

```diff
@@ -116,16 +118,37 @@ public class VbJitCompiler : ScriptingJitCompiler
 {
     public VbJitCompiler(HashSet<Assembly> referencedAssemblies) : base(referencedAssemblies) { }
 
-    protected override int IdentifierKind => (int) SyntaxKind.IdentifierName;
+    protected override int IdentifierKind => (int)SyntaxKind.IdentifierName;
     protected override StringComparer IdentifierNameComparer => StringComparer.OrdinalIgnoreCase;
 
     protected override Script<object> Create(string code, ScriptOptions options) =>
         VisualBasicScript.Create("? " + code, options);
 
     protected override string GetTypeName(Type type) => VisualBasicObjectFormatter.FormatTypeName(type);
 
-    protected override string CreateExpressionCode(string types, string names, string code) =>
-        $"Public Shared Function CreateExpression() As Expression(Of Func(Of {types}))\nReturn Function({names}) ({code})\nEnd Function";
+    protected override string CreateExpressionCode(string types, string names, string code)
+    {
+        string myDelegate = DefineDelegate(types);
+        return $"{myDelegate} \n Public Shared Function CreateExpression() As Expression(Of Func(Of {types}))\nReturn Function({names}) ({code})\nEnd Function";
+    }
+
+    private static string DefineDelegate(string types)
+    {
+        var arrayType = types.Split(",");
+        if (arrayType.Length <= 16) // .net defines Func<TResult>...Funct<T1,...T16,TResult)
+            return String.Empty;
+            
+        var part1 = new StringBuilder();
+        var part2 = new StringBuilder();
+
+        for (var i = 0; i < arrayType.Length - 1; i++)
+        {
+            part1.Append($" In T{i},");
+            part2.Append($" ByVal arg as T{i},");
+        }
+        part2.Remove(part2.Length - 1, 1);
+        return $"Public Delegate Function Func(Of {part1} Out TResult)({part2}) As TResult";
```

Comment:
@mirceaarhip @cosminsandu25 When the workflow is compiled all of the expressions are put into a single class. If there are two expressions that take more than 16 arguments and the arguments have the same types, I think there will be duplicate delegates. Is it possible to give the delegates unique names?